packages/babel-core/src/transformation/normalize-opts.ts

```diff
@@ -30,7 +30,7 @@ export default function normalizeOptions(config: ResolvedConfig): {} {
       sourceType:
         path.extname(filenameRelative) === ".mjs" ? "module" : sourceType,
 
-      sourceFileName: filename,
+      sourceFilename,
       plugins: [],
```

Comment:
This bug, which is the one outlined in https://github.com/babel/babel/issues/13170#issuecomment-869221066 should be fixed in 7.15 though. I will open another PR later.

Reply:
Probably it is also better to avoid setting sourceFilename if it is `"unknown"`